raster-foundry/raster-foundry
# Create cogs in upload processing

## Overview

This PR creates COGs in upload processing and skips ingests.

It also adds an id column to licenses and fetches bands in project edit controller initialization,
since I needed to fix those two things when I was testing with a single band tif (or -- I needed
them to work to get to the problem where we don't handle single band COGs correctly, see
#3617).

### Checklist

- [x] [Swagger specification](https://github.com/raster-foundry/raster-foundry-api-spec) updated (raster-foundry/raster-foundry-api-spec#30)

### Notes

I'm still not getting tiles and I'm not sure why. The COG validates in [the validation tool](http://cog-validate.radiant.earth/html/validate), the file is there, the ingest location points to the correct location in
s3... I'd like to pair with someone to try to figure out why I'm not getting tiles. I think it's probably something obvious if you know the whole list of requirements to make COGs cooperate.

## Testing Instructions

 * rebuild batch container
 * Create an upload of a tif that isn't a cog
 * `rf process-upload <your upload id>`
 * put the created scene in a project
 * check out your sweet new COG

Closes #3597 

Closes #3598 

Closes #3363
